rust/syntect-plugin/src/main.rs

```diff
@@ -192,9 +193,13 @@ impl<'a> Syntect<'a> {
     }
 
     /// Wipes any existing state and starts highlighting with `syntax`.
-    fn do_highlighting(&mut self, view: &mut MyView) {
+    fn do_highlighting(&mut self, view: &mut MyView, lang: Option<&LanguageId>) {
         let initial_state = {
-            let syntax = self.guess_syntax(view.get_path());
+            let language_id = lang.unwrap_or(view.get_language_id());
```

Comment:
Do we ever modify the view's language id? If we don't, then this is going to be using the stale id if `None` is passed, such as after a save event.

Reply:
We do modify it every time, but yes, I think we should remove the `new_lang` parameter to avoid errors in the future and to make the signature simpler.